src/ucp/api/ucp.h

```diff
@@ -1172,6 +1213,56 @@ typedef struct ucp_worker_params {
 } ucp_worker_params_t;
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP performance endpoint attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the performance of a particular endpoint.
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_ep_perf_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Message size to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    size_t            message_size;
+
+    /**
+     * Local memory type to use for determining performance.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t local_memory_type;
+
+    /**
+     * Remote memory type to use for determining performance.
+     * Relevant only for operations that have remote memory access.
+     * This field must be initialized by the caller.
+     */
+    ucs_memory_type_t remote_memory_type;
+
+    /**
+     * Operation type to report performance for.
+     * This field must be initialized by the caller.
+     */
+    ucp_ep_op_type_t  op_type;
+
+    /**
+     * Estimated time (in seconds) required to send a message of a given size
+     * on this endpoint.
+     * This field is set by the @ref ucp_ep_evaluate_perf function.
+     */
+    double            estimated_time;
```

Comment:
@yosefe maybe we should separate in and out arg ? this looks a bit odd. There is also enable mask for this field. What happens if the enable bit is not set but the function is called ?

Reply:
> What happens if the enable bit is not set but the function is called?
Then this field will not be written to by the function